### ISSUE TITLE
Stitching large PNG via arrayjoin: stream error (win)

### ISSUE THREAD

**Issue by Aiosa** [libvips/pyvips#244]
Hi, at first thanks for this awesome lib :)

I have a problem with stitching PNG files to one large file (6000+ images of 2048x2048 resolution). While my code has no problem with smaller images, the large ones give following error:

```
Traceback (most recent call last):
  in main
    final.write_to_file("result.png")
  in write_to_file
    return pyvips.Operation.call(name, self, filename,
  in call
    raise Error('unable to call {0}'.format(operation_name))
pyvips.error.Error: unable to call VipsForeignSavePngFile
  temp/32768-79872.png: unable to open for read
windows error: The program issued a command but the command length is incorrect.

pngload: stream error
vips2png: unable to write to target result.png
```

What I am doing is 
```
images = [pyvips.Image.new_from_file(filename, access="sequential") for filename in imgs]
final = pyvips.Image.arrayjoin(images, across=math.ceil(w/chunk)).crop(0, 0, w, h)
final.write_to_file("result.png")   
```
where chunk is 2048 in this case. The image it fails on is different every time I run it, and I noticed that the PNG file "result.png" is somewhere around 30MB+ when it crashes. All the 6000+ files have together almost 100 MB (there are big areas with no data, uncompressed it has tens of gigs for sure). It's running on windows (64GB ram so the whole image would even uncompressed almost fit there). Also, `32768-79872.png` exists. 

Thanks in advance!

**Comment by jcupitt** [comment]
Hi @Aiosa,

WIndows has a limit of 2048 files open at once which (I think) is very hard to bypass, perhaps you are hitting this?

I think you'll need to assemble the image array in 2000 tile strips, then in a second pass join the strips vertically. PNG format can be very slow, so I would use vips (`".v"`) format for the intermediate files.

**Comment by Aiosa** [comment]
Thank you so much! As I can see legends are true: your speed of response is truly magnificent ( 😄 ). This is the first time I was able to create the image after I've tried almost every image library (except GDAL) out there. Thank you!

It was enough to just increase the chunk size to 4096 which gave me less than 2k images. I think it was also a bit faster as well. I don't expect I will be working with larger images so this will work.
